core/app.go

```diff
@@ -373,3 +376,37 @@ func (a *App) stopCoprocesses() {
 		coprocess.Stop()
 	}
 }
+
+// getStatusHandler generates HTTP response for Telemetry status control plane
+// endpoint.
+func (a *App) getStatusHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test for GET requests
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+
+	servicesJSON, err := json.Marshal(a.Services)
+	if err != nil {
+		log.Errorf("Error generating services for '/status': %v", err)
+	}
+	log.Debugf("Marshaled services: %v", string(servicesJSON))
+
+	// w.Write([]byte(`{ "message": true }`))
+	w.Write(servicesJSON)
+}
+
+// postEnvHandler generates HTTP response for updating environment variables of
+// future CP managed processes.
+func (a *App) postEnvHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test HTTP method
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write([]byte(`{ "message": true }`))
+}
+
+func (a *App) handleServer() *control.Server {
+	s := a.Server
+	s.Mux.HandleFunc("/status", a.getStatusHandler)
+	s.Mux.HandleFunc("/v3/env", a.postEnvHandler)
+	s.Serve()
```

Comment:
Glad I asked, internal events within the main process will definitely help control surface. Maybe even direct the bus straight out of a websocket? ;)

I'll break up what I have and keep future integration in mind for when #290 drops.